huaweicloud/services/smn/resource_huaweicloud_smn_subscription.go

```diff
@@ -62,81 +65,92 @@ func ResourceSubscription() *schema.Resource {
 	}
 }
 
-func resourceSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
+func resourceSubscriptionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
 	client, err := config.SmnV2Client(config.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud smn client: %s", err)
+		return diag.Errorf("error creating smn client: %s", err)
 	}
+
 	topicUrn := d.Get("topic_urn").(string)
 	createOpts := subscriptions.CreateOps{
 		Endpoint: d.Get("endpoint").(string),
 		Protocol: d.Get("protocol").(string),
 		Remark:   d.Get("remark").(string),
 	}
-	logp.Printf("[DEBUG] Create Options: %#v", createOpts)
+	log.Printf("[DEBUG] Create Options: %#v", createOpts)
 
 	subscription, err := subscriptions.Create(client, createOpts, topicUrn).Extract()
 	if err != nil {
-		return fmtp.Errorf("Error getting subscription from result: %s", err)
-	}
-	logp.Printf("[DEBUG] Create : subscription.SubscriptionUrn %s", subscription.SubscriptionUrn)
-	if subscription.SubscriptionUrn != "" {
-		d.SetId(subscription.SubscriptionUrn)
-		d.Set("subscription_urn", subscription.SubscriptionUrn)
-		return resourceSubscriptionRead(d, meta)
+		return diag.Errorf("error getting subscription from result: %s", err)
 	}
+	log.Printf("[DEBUG] Create: subscription.SubscriptionUrn %s", subscription.SubscriptionUrn)
 
-	return fmtp.Errorf("Unexpected conversion error in resourceSubscriptionCreate.")
+	d.SetId(subscription.SubscriptionUrn)
+	return resourceSubscriptionRead(ctx, d, meta)
 }
 
-func resourceSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+func resourceSubscriptionRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
 	client, err := config.SmnV2Client(config.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud smn client: %s", err)
+		return diag.Errorf("error creating smn client: %s", err)
 	}
+	id := d.Id()
 
-	logp.Printf("[DEBUG] Deleting subscription %s", d.Id())
+	log.Printf("[DEBUG] Getting subscription %s", id)
 
-	id := d.Id()
-	result := subscriptions.Delete(client, id)
-	if result.Err != nil {
-		return result.Err
+	subscriptionslist, err := subscriptions.List(client).Extract()
```

Comment:
s/subscriptions.List/subscriptions.ListFromTopic(client, topicUrn)
from the console , actual: Subscription urn = {topic urn}:UUID